### PR TITLE
Update workflow instructions

### PR DESCRIPTION
**Pull request by MatthiasHeilManchester**
Draft code review policy added to CONTRIBUTING.md.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -255,6 +256,11 @@ This involves the following steps:
    will pull these formatting changes back into your local repository. To avoid merge conflicts
    it's a good habit to do this a few minutes after pushing anything to your remote repository.
 
+   Also remember that a push to a remote repository will usually spawn a number of Action-based self-tests. While these don't "cost" us anything (Thank you, Github!) you should still kill unnecessary runs (in the Actions tab) if the push to your forked repository is intended as a backup and/or has not affected parts of the library that are tested by these Actions. For instance, there's no need to re-run the self-tests if you've only worked on the documentation. You can cancel a particular workflow by clicking on the "..." button as shown here:
```
Comment (jondea) [inline CONTRIBUTING.md:259]: > and/or has not affected parts of the library that are tested by these Actions. For instance, there's no need to re-run the self-tests if you've only worked on the documentation.

I would weakly disagree with this, I find that it's easy to convince myself that I definitely haven't broken something when I possibly have. It also takes mental load off reviewers to see all green before merging. Saying this, I agree with the general point though: if no one is ever going to care about the result, it's good to cancel it.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -268,6 +274,10 @@ This involves the following steps:
 
     ![](doc/README/reviewers_button.png)
 
+    Also choose one or more labels(s) from the dropdown menu to indicate which part of the library is affected by your pull request.
```
Comment (jondea) [inline CONTRIBUTING.md:277]: labels(s) -> label(s)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -338,6 +348,45 @@ This involves the following steps:
...
+The general workflow for approving a pull request is as follows:
+
+1. **Reviewer:** Go to the "Files changed" tab for the pull request
+
+      ![](doc/README/files_changed_tab.png)
+
+   and inspect the changes made.
+
+ 1. **Reviewer:** Add comments/requests for change by leaving a comment near the relevant part of the code. Hover over the line number until the blue plus sign appears
+    ![](doc/README/plus_sign.png)
```
Comment (jondea) [inline CONTRIBUTING.md:364]: This picture and several after it are appearing inline and [it looks a bit odd](https://github.com/MatthiasHeilManchester/oomph-lib/blob/update_workflow_instructions/CONTRIBUTING.md#code-review). If you add a newline to separate the text and image, it should make it display not inline.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -338,6 +348,45 @@ This involves the following steps:
...
+    
+1. **Reviewer:** Keep going until you've reviewed all the changed files and left comments on all relevant issues, then finish the review by pressing the green "Finish your review" button at the very top of the page.
+      ![](doc/README/finish_your_review.png)
+
+1. **Reviewer:** Add a brief overall summary of the issues raised in the window that appears:
+      ![](doc/README/finish_review2.png)
+
+   If the changes are deemed essential, check the "Request changes" radio button before submitting the review.
+
+1. **Reviewee:** Work your way through the comments and either change the code as suggested (do this in your local cloned version of your forked repository, and then commit/push the changes to your remote forked repository; they will then automatically be included into the pull request), and leave a comment to say that you've done so. Alternatively, discuss/clarify the issue with the reviewer.
```
Comment (jondea) [inline CONTRIBUTING.md:377]: > then commit/push the changes to your remote forked repository

might be worth adding that the commits should be on the same branch the the PR was originally made from

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -338,6 +348,45 @@ This involves the following steps:
...
+
+1. **Reviewee:** When you believe the issue has been resolved to everybody's satisfaction (this may involve a few iterations), leave a final comment "Done" in the chain of comments, and let the reviewer know when you've dealt with all of them.
+
+1. **Reviewer:** Revisit the issues. Sadly, there does not seem to be an easy way to do this; you'll have to scroll back through the conversation page to find any change requests that you made. They are marked with a red symbol:
+![](doc/README/resolve_conversation.png)
+Once you're happy with the changes made (or agree that the request wasn't sensible), press the "Resolve conversation" button. 
+
+1. **Reviewer:** Once all the issues have been dealt with to your satisfaction, go to the bottom of the page and indicate that you approve the changes made as part of this pull request:
+  ![](doc/README/list_of_changes_requested_approve.png)
```
Comment (jondea) [inline CONTRIBUTING.md:386]: Looks like you've got some interesting reading on those other tabs ;)

**Review by jondea — COMMENTED** [review]
Looks good! Just some minor comments

File: CONTRIBUTING.md (hunk 1 above)
Comment (MatthiasHeilManchester) [reply CONTRIBUTING.md:259]: Slightly rephrased to make it clear that a full set of self-tests should be run before submitting a pull request.

File: CONTRIBUTING.md (hunk 2 above)
Comment (MatthiasHeilManchester) [reply CONTRIBUTING.md:277]: Done!

File: CONTRIBUTING.md (hunk 3 above)
Comment (MatthiasHeilManchester) [reply CONTRIBUTING.md:367]: Done

**Pull request merged**